//
// {C} Copyright 2020 Pensando Systems Inc. All rights reserved
//
//------------------------------------------------------------------------------
///
/// \file
/// protobuf specification for syslog configuration
///
//------------------------------------------------------------------------------

syntax = "proto3";
package operd;

import "gogo.proto";
import "meta/meta.proto";
import "types.proto";

service SyslogSvc {
  rpc SetSyslogConfig (SyslogConfigRequest) returns (SyslogConfigResponse) {}
}

enum SyslogProtocol {
  BSD = 0;
  RFC = 1;
}

message SyslogConfig {
  string ConfigName       = 1;
  string RemoteAddr       = 2;
  uint32 RemotePort       = 3;
  SyslogProtocol Protocol = 4;
  uint32 Facility         = 5;
  string Hostname         = 6;
  string AppName          = 7;
  string ProcID           = 8;
}

message SyslogConfigRequest {
  SyslogConfig Config = 1;
}

message SyslogConfigResponse {
  types.ApiStatus ApiStatus = 1;
}